### DISCUSSION TITLE
How to detect the change of palettes?

### DISCUSSION THREAD

**Discussion by eyllanesc** [squidfunk/mkdocs-material#3429] [Q&A] [answered]
I know I can detect the [change of palette](https://squidfunk.github.io/mkdocs-material/setup/changing-the-colors/?h=dark#color-palette-toggle) using MutationObserver but it seems to me that it is a hack, is there a stable and official way to do it? Any event that provides mkdocs-material?

**Comment by squidfunk** [comment, marked as answer]
This should work:

``` js
const ref = document.querySelector("[data-md-component=palette]")
component$.subscribe(component => {
  if (component.ref === ref) {
    alert$.next(`Changed to "${component.color.scheme}"`)
    console.log(component.index, component.color)
  }
})
```

In general you can monitor all components through the `component$` observable. There's also the `document$` observable to monitor document changes (instant loading) and `location$` to track location changes, and more:

https://github.com/squidfunk/mkdocs-material/blob/12a016bcf9658cb782af6fffed1a17c0f474cc78/src/assets/javascripts/bundle.ts#L249-L258

**Reply by eyllanesc** [reply]
It would be great if this type of information is documented.
